### PR TITLE
Implemented custom skeletonkey Namespace object and enabled pretty-printing

### PR DESCRIPTION
**Pull request by DylanScottCarroll**
You don't have to merge this, I just had fun writing it lol, so I jumped the gun.

I created a new custom Namespace class in configs.py.  The Namespace class is essentially just the same as argparse.Namespace but has __getitem__, __setitem__, and __str__ methods. It can also take 

I replaced all references to argparse.Namespace with the skeletonkey custom Namespace class. 

The __str__ method formats the nested namespaces in a hierarchical way essentially in yaml format.

I also have a change in here that pulls out some of the keywords as constants at the top of the file. I meant that to be a separate pull request, but I was working on the wrong branch and don't know how to detangle the git history, so I'm leaving it in. If you don't like that change you can reject this pull request and I'll remove that bit.

### CODE REVIEW

**Comment by DylanScottCarroll** [comment]
I changed the name "Namespace" to key for branding and to avoid confusion.

File: skeletonkey/config.py [hunk 1]
```diff
@@ -282,34 +369,18 @@ def add_args_from_dict(
...
-        namespace (argparse.Namespace): The Namespace object to be converted.
+        key (Key): The Key object to be converted.
 
     Returns:
-        argparse.Namespace: A nested Namespace representation of the input Namespace object.
+        Key: A nested Key representation of the input Key object.
     """
     nested_dict = {}
-    for key, value in vars(namespace).items():
+    for key, value in vars(key).items():
```
Comment (sizemore0125) [inline skeletonkey/config.py:383]: This feels weird. I have feeling that this might cause issues. Maybe we should use a different name other that "key".

File: skeletonkey/core.py [hunk 2]
```diff
@@ -9,9 +9,11 @@
     load_yaml_config,
     add_args_from_dict,
     add_yaml_extension,
-    namespace_to_nested_namespace,
+    key_to_nested_key,
+    Key
 )
 
+TARGET_KEYWORD: str = "_target_"
```
Comment (sizemore0125) [inline skeletonkey/core.py:16]: I like these changes, but I feel like this should be a separate PR.

File: skeletonkey/core.py [hunk 3]
```diff
@@ -118,10 +120,9 @@ def instantiate(namespace: argparse.Namespace, **kwargs) -> Any:
     Raises:
         TypeError: If the class is missing specific parameters.
     """
-    obj_kwargs = vars(namespace).copy()
-    target_keyword = "_target_"
-    class_obj = import_class(obj_kwargs[target_keyword])
-    del obj_kwargs[target_keyword]
+    obj_kwargs = vars(key).copy()
+    class_obj = import_class(obj_kwargs[TARGET_KEYWORD])
+    del obj_kwargs[TARGET_KEYWORD]
```
Comment (sizemore0125) [inline skeletonkey/core.py:125]: I am not a fan of the pattern where you use a global variable in the body of a function. Instead it should be used as the default parameter in a function. One future feature I would like to have is the ability to disable skeletonkey features from "unlock", and this pattern would make it hard.

File: skeletonkey/config.py (hunk 1 above)
Comment (sizemore0125) [reply skeletonkey/config.py:383]: Maybe "ConfigVault" or something like that would be better. (And metaphorically consistent).

File: skeletonkey/core.py (hunk 2 above)
Comment (DylanScottCarroll) [reply skeletonkey/core.py:16]: I agree. I intended to make them as part of a separate PR, but I was accidentally working on the wrong branch of my repo when I added the new class and didn't think it was worth it to try to detangle the changes to make them separate PRs. In the future, I'd make sure they were separate.

File: skeletonkey/core.py (hunk 3 above)
Comment (DylanScottCarroll) [reply skeletonkey/core.py]: Ah, I'd never thought of that! That's a good pattern to stick to.

**Comment by DylanScottCarroll** [comment]
I believe that I addressed all of the concerns, and I pushed the updated changes!

File: README.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ For instance, if your configuration file has a nested YAML, you can overwrite th
 python project.py --model.parameters.layer_size 256
 ```
 
-The resulting Namespace object will contain nested Namespace objects that can be accessed using dot notation, such as args.model.parameters.layer_size.
+The resulting Key object will contain nested Key objects that can be accessed using dot notation, such as args.model.parameters.layer_size.
```
Comment (sizemore0125) [inline README.md:66]: Documentation was not updated in your PR.

File: example/basic_example/main.py [hunk 5]
```diff
@@ -1,3 +1,7 @@
+# Adds the local skeletonkey source code to path, so that version is imported
```
Comment (sizemore0125) [inline example/basic_example/main.py:1]: I am open to this change. But I think this should be a separate PR.

**Review by sizemore0125 — CHANGES_REQUESTED** [review]

**Comment by DylanScottCarroll** [comment]
Alright, I think I fixed those two things.

**Review by sizemore0125 — APPROVED** [review]
Good Job!
